src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraph/XmlPlanNodeBuilder.cs

```diff
@@ -371,6 +374,100 @@ private IEnumerable<BaseStmtInfoType> EnumStatements(StmtBlockType statementBloc
             }
         }
 
+        private Description ParseDescription(ShowPlanGraph graph, string XmlDocument, int index)
+        {
+
+            Description description = new Description();
+            description.HasMissingIndex = true;
+            description.QueryText = graph.Statement;
+
+            XmlDocument stmtXmlDocument = new XmlDocument();
+            stmtXmlDocument.LoadXml(XmlDocument);
+            var nsmgr = new XmlNamespaceManager(stmtXmlDocument.NameTable);
+            nsmgr.AddNamespace("shp", "http://schemas.microsoft.com/sqlserver/2004/07/showplan");
+            XmlNode rootNode = stmtXmlDocument.DocumentElement;
+
+            XmlNode missingIndexes = rootNode.SelectSingleNode("descendant::shp:MissingIndexes", nsmgr);
+            //Returning null if no missing indexes are found in the xml file
+            if (missingIndexes == null)
+            {
+                return null;
+            }
+            XmlNodeList indexGroups = missingIndexes.SelectNodes("descendant::shp:MissingIndexGroup", nsmgr);
+
+            bool memoryOptimzed = false;
+            XmlNode scan = rootNode.SelectSingleNode("descendant::shp:IndexScan", nsmgr);
```

Comment:
Where is all this logic from? Some explanation of what is going on would be helpful...this is really hard to understand currently without expertise on how showplan is set up.